nncf/torch/sparsity/movement/algo.py

```diff
@@ -0,0 +1,652 @@
+        with open(os.path.join(dirname, 'structured_sparsity.md'), 'w') as f:
+            df.to_markdown(f)
+
+
+    @property
+    def compression_rate(self):
+        return self.statistics().movement_sparsity.model_statistics.sparsity_level
+
+    def prepare_for_export(self):
+        """
+        Applies pruning masks to layer weights before exporting the model to ONNX.
+        """
+        self._propagate_masks()
+
+    def _propagate_masks(self):
+        def calc_sparsity(tensor):
+            return 1-tensor.count_nonzero()/tensor.numel()
+        # nncf_logger.debug("MVMT - Propagating pruning masks")
+        # 1. Propagate masks for all modules
+        from collections import OrderedDict
+        sparse_sd = OrderedDict()
+        with torch.no_grad():    
+            for sparse_info in self.sparsified_module_info:
+                for n, m in self.model.named_modules():
+                    if m == sparse_info.module:
+                        # print("- SparseModule: {} -".format(n))
+                        # print("\tw_mask sparsity: {:.3f}".format(calc_sparsity(sparse_info.operand.weight_ctx.binary_mask)))
+                        # print("\tw_sd   sparsity: {:.3f}".format(calc_sparsity(m.weight)))
+                        sparse_sd[n+'.weight'] = sparse_info.operand.apply_binary_mask(m.weight)
+                        # print("\t*w_sd  sparsity: {:.3f}".format(calc_sparsity(sparse_sd[n+'.weight'])))
+
+                        if hasattr(m, 'bias'):
+                            # print("\tb_mask sparsity: {:.3f}".format(calc_sparsity(sparse_info.operand.bias_ctx.binary_mask)))
+                            # print("\tb_sd   sparsity: {:.3f}".format(calc_sparsity(m.bias)))
+                            sparse_sd[n+'.bias'] = sparse_info.operand.apply_binary_mask(m.bias, isbias=True)
+                            # print("\t*w_sd  sparsity: {:.3f}".format(calc_sparsity(sparse_sd[n+'.bias'])))
+
+        model_sd = self.model.state_dict()
+        for k, v in sparse_sd.items():
+            assert k in model_sd, "key not exists!"
+            model_sd[k] = sparse_sd[k]
+        self.model.load_state_dict(model_sd)
+
+    def print_prunableops_per_group(self):
+        for group, op_list in self.prunableops_per_group.items():
+            print("= Group {} ======".format(group))
+            print('\n'.join(list(map(lambda x: '{:12} | {}'.format(str(list(x.op_mod.weight.shape)), str(x.op_addr)), op_list))))
+  
+    def _get_group_of_prunable_ops(self):
+        PrunableOp = namedtuple("PrunableOp", "op_addr op_mod")
+
+        building_blocks, _ = get_building_blocks(self.model,
+                                target_block_types=[BuildingBlockType.MSHA, BuildingBlockType.FF],
+                                block_filter_strategy=BlockFilteringStrategy.KEEP_SMALL,
+                                hw_fused_ops=True)
+
+        all_node_op_addr_in_blocks = self._get_all_node_op_addresses_in_block(self.model, building_blocks)
```

Comment:
Seems like it's not needed now since `get_building_blocks` returns `ExtendedBuildingBlock`. It has `op_addresses`.

Reply:
The only dependency of the caller is ```create_structured_sparsity_context``` which we are revising now. If it ends up mandatory, we will refactor to use `op_addresses` of `ExtendedBuildingBlock`